messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -1076,8 +1075,9 @@ public void run() {
                 // Submit segmentation workers matching the size of our thread pool (-1 for the current dispatcher).
                 // Keep track of the last processed segments...
                 TrackingSegmentWorker workingInCurrentThread = null;
-                for (int i = 0; i < tokenStoreCurrentSegments.length && availableThreads.get() > 0; i++) {
-                    int segmentId = tokenStoreCurrentSegments[i];
+                int[] availableSegments = tokenStore.fetchAvailableSegments(processorName).orElse(tokenStore.fetchSegments(processorName));
```

Comment:
Strictly speaking, this wouldn't be the `availableSegments` if the `orElse` is invoked.
However, that would require the naming to switch depending on the operation...also not very helpful.

Reply:
Good point, I'll change it into `segmentsToClaim`, since that's basically what we'll try with them in either scenario